compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1533,6 +1541,12 @@ impl<'tcx> PlaceRef<'tcx> {
         }
     }
 
+    /// If MirPhase >= Derefered and if projection contains Deref,
+    /// It's guaranteed to be in the first place
+    pub fn has_deref(&self) -> bool {
+        self.projection.first() == Some(&PlaceElem::Deref)
```

Comment:
The same debug assertion should also be added here -- right now it seems very easy to accidentally cal `has_deref` before the "deref only in first projection" invariant has been established.

Reply:
Added the debug assertion and bit better documentation here https://github.com/rust-lang/rust/pull/114505